### DISCUSSION TITLE
CAN protocol  of pylontech compatible with Taico

### DISCUSSION THREAD

**Discussion by hochers** [hoylabs/OpenDTU-OnBattery#519] [General]
Hi, i am thinking about buying a Taico tk2400 lifepo4 battery. It looks pretty much identical to the pylontech us2000. Does anybody know if the protocol is standardized and if it might work with openDTU-onBattery? Thank you

**Comment by hochers** [comment]
Also ich habe nun die Taico TK (oder TS 2400) Batterie, und habe sie versucht ähnlich zur Pylontech mit dem Bus transceiver SN65HVD230  am ESP anzuschließen. Ich hab per RS232 das BMS auf CAN Bus "Pylontech" umgestellt. Leider bekomme ich keine Verbindung mit openDTU. (PIN 4 CANH, PIN5 CANL, PIN2 GND)  Hat jemand noch eine Idee?
![image](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/20065904/e1c92ba1-00ef-4ace-91ac-a450debee4d4)

**Comment by Manos1966** [comment]
Hallo @hochers 
ich glaube diese Frage muss du an den Kundendienst von Taico stellen.

Alles andere hilft nichts. Es ist als ob du einen Windows Rechner und einen Linix PC zusammen verbinden willst, weil beide den gleichen LAN RJ-45 Stecker haben...

Anscheinend viele Batterie Hersteller benutzen das Pylontech Protokoll. 
Der Taico Kundendienst wird wissen ob das gleiche Protokoll benutzt wird wie Pylontech.

**Comment by hochers** [comment]
Hallo @Manos1966 ,

Danke für deine Antwort, in der BMS Software kann ich anscheinend den CAN Bus auf das Pylontech Protokoll einstellen. Habe nun auch bei Taico angefragt, mal abwarten was die antworten. Könnte es auch an mir liegen? Habe verschiede Anschlussvarianten (mit und ohne GND etc) versucht und auch mit und ohne Abschlusswiderstand. 

Danke und lg

**Reply by Manos1966** [reply]
Na ja, es "konnte an dir lieger" da du keine Moeglichkeit hast zu pruefen, ob deine Canbus Installation richtig ist.

Dein |System kannst du aber testen, wenn du ein Raspberry Pi hast:
[pylontech-emulator](https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Pylontech-battery#pylontech-emulator)

**Reply by hochers** [reply]
Hi @Manos1966,

leider bislang alles vergebens, TAICO support meint CAN Bus sollte funktionieren und das Setting "Pylontech" ist laut Aussage für den Inverter gedacht. Das bedeutet ja, dass es mit openDTU funktionieren sollte. In der CAN Bus Beschreibung die mir übermittelt wurde steht leider keine Beschreibung zur "pylontech" Einstellung, und die aufgeführten Registeradressen sind anders als bei Pylontech
[Taico_CAN_en.pdf](https://github.com/helgeerbe/OpenDTU-OnBattery/files/13316847/Taico_CAN_en.pdf)

**Comment by hochers** [comment]
Hi ihr,
ich habe mir ein System mit OpenDTU-onBattery aufgebaut. Ich habe eine TAICO Lifepo4 (TS2400) Battery und würde sie gerne mit einbinden. Der CAN Bus lässt sich zwar auf "pylontech" einstellen, aber es gibt keine Verbindung. Ich habe schon verschiedene CAN Bus Breakout boards versucht und auch mit und ohne Abschlusswiderstand. Das TAICo BMS Tool mit RS232 funktioniert ohne Probleme... Ich habe von TAICO folgende Informationen bekommen und habe versucht zuerst mal in Python eine Kommunikation mit RS232 herzustellen, leider ohne Erfolg, verstehe4 das Protokoll nicht so ganz. Gibt es hier jemanden der sich damit auskennt und mit in die richtige Richtung schicken kann ;). 
[Taico_CAN_en.pdf](https://github.com/helgeerbe/OpenDTU-OnBattery/files/13331296/Taico_CAN_en.pdf)
[MODBUS[1].pdf](https://github.com/helgeerbe/OpenDTU-OnBattery/files/13331303/MODBUS.1.pdf)

Danke
